dhall/src/Dhall.hs

```diff
@@ -96,13 +99,16 @@ module Dhall
     , Generic
     ) where
 
-import Control.Applicative (empty, liftA2, (<|>), Alternative)
+import Control.Applicative (empty, liftA2, Alternative)
 import Control.Exception (Exception)
 import Control.Monad.Trans.State.Strict
 import Control.Monad (guard)
 import Data.Coerce (coerce)
+import Data.Either.Validation (Validation(..))
+import Data.Functor.Alt ((<!>))
```

Comment:
I believe you can use `<|>` instead of `<!>`, that way you don't need to directly depend on `semigroupoids`

Reply:
I've replaced `<|>` with `<!>` because it requires a `Monoid`, whereas `NonEmpty a` (and `Validation (NonEmpty a) b`) only have a `Semigroup`. If the dependency on semigroupoids is a problem, I can just reimplement `<!>` instead of pulling the library.

Reply:
Oh, alright.  Then you can leave the `(<!>)`.  Minimizing the constraint is a good reason to use it